src/main/java/pro/java/hw15/ValueCalculator.java

```
@@ -0,0 +1,54 @@
+package pro.java.hw15;
+
+import java.util.Arrays;
+
+public class ValueCalculator {
+    private final double[] values;
+    private final int halfSize;
+
+    public ValueCalculator(int arraySize) {
+        int arraySize1 = Math.max(arraySize, 1000000);
+        this.halfSize = arraySize1 / 2;
+        this.values = new double[arraySize1];
+    }
+
+    public void fillAndSplitArray() {
+        long start = System.currentTimeMillis();
+
+        Arrays.fill(values, 1);
+
+        double[] a1 = new double[halfSize];
+        double[] a2 = new double[halfSize];
+
+        System.arraycopy(values, 0, a1, 0, halfSize);
+        System.arraycopy(values, halfSize, a2, 0, halfSize);
+
+        Thread thread1 = new Thread(() -> calculateValues(a1, 0));
+        Thread thread2 = new Thread(() -> calculateValues(a2, halfSize));
+
+        thread1.start();
+        thread2.start();
+
+        try {
+            thread1.join();
+            thread2.join();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Better to throw an exception to inform somehow what was happened